### PR TITLE
make macOS package use a consistent path

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

Address mac portions of https://github.com/PowerShell/PowerShell-RFC/pull/115#issuecomment-394542248

- Make path when stable `/usr/local/microsoft/powershell/6`
  - when not stable (preview) `/usr/local/microsoft/powershell/6-preview` and symlink name pwsh-preview
- allow side-by-side install of stable and preview

Also:
- fix an issue where the `utime` work around for mac requires sudo
- refactor some code into common functions

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -2212,6 +2248,24 @@ function New-MSIPatch
...
 }
 
+<#
+    .Synopsis
+        Tests if a version is preview
+    .EXAMPLE
+        Test-IsPreview -version '6.1.0-sometthing' # returns true
+        Test-IsPreview -version '6.1.0' # returns false
+#>
+function Test-IsPreview
```
Comment (rjmholt) [inline tools/packaging/packaging.psm1]: Only note here is that a string like `"6.0.0-alpha.13-1557-gd67d87c"` also passes this test. Just want to make sure that's intended behaviour (that string gets caught by the validation attribute so not a big concern).

**Review by rjmholt — APPROVED** [review]
I've tested this out on Ubuntu 16.04 (i.e. package and install) and it works.

Tested as non-sudo, and package-install-package path. Works well.

Code changes look good.

File: tools/packaging/packaging.psm1 (hunk 1 above)
Comment (TravisEz13) [reply tools/packaging/packaging.psm1]: an alpha would be considered a preview.  basically anything with a dash (`-`) in the version is a preview.

**Comment by TravisEz13** [comment]
removing breaking change as the breaking part was made in a previous change.

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Pull request merged**